routing/payment_lifecycle.go

```diff
@@ -363,40 +363,25 @@ func (p *paymentLifecycle) requestRoute(
 	log.Warnf("Failed to find route for payment %v: %v", p.identifier, err)
 
 	// If the error belongs to `noRouteError` set, it means a non-critical
-	// error has happened during path finding and we might be able to find
-	// another route during next HTLC attempt. Otherwise, we'll return the
-	// critical error found.
+	// error has happened during path finding and we will mark the payment
+	// failed with this reason. Otherwise, we'll return the critical error
+	// found to abort the lifecycle.
```

Comment:
we don't do so because it indicates a non-routing related error, like bugs in other subsystems which have nothing to do with payment routing, hence the payment could have been succeeded given its amount, its params such as fees or locktimes, and the current graph.

Reply:
SGTM